src/components/ButtonGroup/ButtonGroup.js

```diff
@@ -0,0 +1,74 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import styles from './style.scss';
+
+const baseClass = 'btn-group';
+
+class ButtonGroup extends React.Component {
+  state = {
+    currentIndex: this.props.currentIndex || -1
+  };
+
+  makeClickHandler = index => event => {
+    this.props.onChange(event, index);
+    this.setState({ currentIndex: index });
+  };
+
+  render() {
+    const { size, fullWidth, children, ...restProps } = this.props;
+    const currentIndex =
+      typeof this.props.currentIndex === 'number'
+        ? this.props.currentIndex
+        : this.state.currentIndex;
+
+    const mappedChildren = React.Children.map(children, (child, index) =>
+      React.cloneElement(child, {
+        size,
+        fullWidth,
+        primary: false,
+        destructive: false,
+        onClick: this.makeClickHandler(index),
+        className: index === currentIndex ? styles.active : null
+      })
+    );
+
+    return (
+      <div role="group" className={styles[baseClass]} {...restProps}>
+        {mappedChildren}
+      </div>
+    );
+  }
+}
+
+ButtonGroup.defaultProps = {
+  onChange: () => {}
+};
+
+ButtonGroup.propTypes = {
+  /**
+   * If `true`, the buttons in group will take up the full width of its container
+   */
+  fullWidth: PropTypes.bool,
```

Comment:
Maybe we should pass `className` instead of `fullWidth`?

Reply:
I would like to allow users to use the same props like the ones on `Button` component because after all they are assigned to those buttons rendered inside the group while rendering. But if you have in mind a better way of handling this behaviour  I am open for suggestions 🙂

Reply:
I think we should provide `className` and `fullWidth`. As I see both are used in `Button` component.

Reply:
ok, I have updated the component to use also `className` 👍